### PR TITLE
Finish transforming target/arduino_101 to target/zephyr

### PR DESCRIPTION
**Pull request by pfalcon**
This is follow up patch to https://github.com/Samsung/jerryscript/pull/1143 , and based on the suggestion of the original contributor, @sergioamr , that if more than one board of Zephyr is supported, that it makes sense to rename target to just "zephyr".

Taking a comment from the previous PR:

> I am a bit surprised (confused) because I thought Zephyr is x86 only.

- Zephyr is actually quite portable and supports number of architectures. To prove that my changes from #1143 are usable beyond just x86 and ARM, I've added an ARC architecture board. I don't have one, so cannot test if it boots, but it builds an ELF which has the correct machine code. I guess it's helpful to include (with a comment that it's not tested fully).

Then, the patchset goes over renaming files/dirs and updating README to mention support for more boards of different architectures (well, so far it's mostly "virtual" boards of qemu_x86 and qemu_cortex_m3).

As before, there're individual commits for ease of review, ready to be squashed once overall approach is ack'ed.

@sergioamr : Please review.

### CODE REVIEW

File: targets/zephyr/README.md [hunk 1]
```diff
@@ -19,25 +18,26 @@ The folder tree related would look like this.
 harmony
   + jerry
```
Comment (lemmaa) [inline targets/zephyr/README.md]: Better to use 'jerryscript' instead 'jerry'. :)

File: targets/zephyr/README.md [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
+run on a number of supported boards (like
+[Arduino 101 / Genuino 101](https://www.arduino.cc/en/Main/ArduinoBoard101),
+[Zephyr Arduino 101](https://www.zephyrproject.org/doc/board/arduino_101.html)).
 
 ### How to build
 
 #### 1. Preface
 
-1, Directory structure
+1. Directory structure
```
Comment (zherczeg) [inline targets/zephyr/README.md]: I cannot figure this out, but '1' is deleted?

File: targets/zephyr/README.md [hunk 3]
```diff
@@ -19,25 +18,26 @@ The folder tree related would look like this.
...
   + zephyr
 ```
 
 
-2, Target board
+2. Target boards/emulations
 
-Assume [Arduino 101 / Genuino 101](https://www.arduino.cc/en/Main/ArduinoBoard101)
-as the target board.
+Following Zephyr boards were tested: qemu_x86, qemy_cortex_m3, arduino_101.
```
Comment (zherczeg) [inline targets/zephyr/README.md]: qemu_cortex_m3

**Comment by zherczeg** [comment]
I think the patch is OK in general. Just a few fixes

File: targets/zephyr/README.md (hunk 2 above)
Comment (poussa) [inline targets/zephyr/README.md]: , is changed to . which is OK

**Comment by poussa** [comment]
LGTM

**Comment by sergioamr** [comment]
Finally I've got a computer back and i can look at this. 
Please, have a look to my changes on your branch.

**Comment by sergioamr** [comment]
@pfalcon please check : https://github.com/pfalcon/jerryscript/pull/1

**Comment by pfalcon** [comment]
@sergioamr : Thanks, I'm going to integrate it later today. As everything should be squashed together I guess, there will be just a signoff line from you, is it ok?

File: targets/zephyr/README.md (hunk 1 above)
Comment (pfalcon) [inline targets/zephyr/README.md]: Fixed.

File: targets/zephyr/README.md (hunk 3 above)
Comment (pfalcon) [inline targets/zephyr/README.md]: Fixed.

**Comment by pfalcon** [comment]
All suggestions and additional patches integrated, squashed to a single commit, rebased to the master as of now.

**Comment by akosthekiss** [comment]
@pfalcon travis signals error now. Please swap the signed-off lines. As you are commiting the patch, the checker script expects your signed-off line as the last one in the commit message.

**Comment by pfalcon** [comment]
Thanks for the hint, done. Watching Travis...

**Comment by sergioamr** [comment]
LGTM.

**Comment by zherczeg** [comment]
LGTM

**Comment by LaszloLango** [comment]
LGTM, please rabase to the current master, so we can land it.

**Comment by poussa** [comment]
@LaszloLango why the rebase is needed? There is no conflicts.

@pfalcon if there is indeed something that needs to be done, can you follow up so we get this landed.

**Comment by LaszloLango** [comment]
@poussa, I know there is no conflicts. The "problem" is in our development process. We don't use merge commits, so we have to rebase the master to the branch we want to merge and push. Branch must be up to date to do this. But it changes the hash of the commit, so GitHub won't notice it is on master when we pushed it. I can land the PR without rebasing it, but I have to close the PR manually after it's done. Other option is to use the 'squash and merge' option of GitHub. It does exactly what we need, but it changes the author of the commit (based on GitHub user info) and usually breaks the Sign-off checker, so travis (and `make precommit`) will fail.

**Comment by poussa** [comment]
@LaszloLango Understood. Lets follow the process then.

**Comment by pfalcon** [comment]
Rebased - sorry for the delay, I'm on vacation this week, may be not available next days again. @poussa : if there is a need to rebase again, maybe you could clone this PR and rebased/resubmit it? Thanks in advance.

**Pull request merged**